frontend/server/src/Controllers/Run.php

```diff
@@ -913,18 +940,38 @@ public static function apiSource(\OmegaUp\Request $r): array {
 
         return self::getOptionalRunDetails(
             $submission,
-            $run,
-            false
+            $run
         );
     }
 
+    private static function shouldShowRunDetails(
+        int $identityId,
+        \OmegaUp\DAO\VO\Problems $problem,
+        ?\OmegaUp\DAO\VO\Contests $contest
+    ): string {
+        $isProblemSolved = \OmegaUp\DAO\Problems::isProblemSolved(
+            $problem,
+            $identityId
+        );
+        if (
+            !is_null($contest)
+            && (
+                !$isProblemSolved
+                || $contest->finish_time > \OmegaUp\Time::get()
+            )
+        ) {
+            return $contest->feedback;
+        }
+        return $isProblemSolved ? 'detailed' : 'none';
+    }
+
     /**
      * @return array{compile_error?: string, details?: array{compile_meta?: array<string, array{memory: float, sys_time: float, time: float, verdict: string, wall_time: float}>, contest_score: float, groups?: list<array{cases: list<array{contest_score: float, max_score: float, meta: array{verdict: string}, name: string, score: float, verdict: string}>, contest_score: float, group: string, max_score: float, score: float}>, judged_by: string, max_score?: float, memory?: float, score: float, time?: float, verdict: string, wall_time?: float}, source: string}
      */
     private static function getOptionalRunDetails(
         \OmegaUp\DAO\VO\Submissions $submission,
         \OmegaUp\DAO\VO\Runs $run,
-        bool $showDetails
+        bool $showDetails = false
```

Comment:
```suggestion
        bool $showDetails
```

creo que prefiero que este parámetro siempre sea explícito.

Reply:
Listo